### DISCUSSION TITLE
Exampe: Most simple countdown

### DISCUSSION THREAD

**Discussion by pke** [statelyai/xstate#1729] [Q&A]
I tried to create a simple countdown chart, but somehow its not counting down:

```js
const guards = {
  canCountDown: context => !guards.countedDown(context),
  countedDown: context => context.count === 0
}

const countdownMachine = Machine({
  id: "countdown",
  initial: "idle",
  context: {
    count: 10
  },
  states: {
    idle: {
      on: {
        START: {
          actions: "startCounting",
          target: "counting"
        }
      }
    },
    counting: {
      after: {
        1000: {
          target: "counting",
          actions: "countdown",
          cond: "canCountDown"
        }
      },
      on: {
        "": { 
          target: "done",
          cond: "countedDown"
        },
        STOP: {
          target: "idle",
          actions: "reset"
        }
      }
    },
    done: {
      type: "final",
      on: {
        RESTART: {
          target: "counting",
          actions: "reset"
        }
      }
    }
  }
}, {
  guards,
  startCounting: assign({ startCount: context => context.count }),
  reset: assign({ count: context => context.startCount }),
  countdown: assign({ count: context => context.count - 1 })
}) 
```

https://xstate.js.org/viz/?code=0c13d2e2de2cd30bbc2b&gist=9ce4bb4840c35b4c3567adf98dacf8fb

**Comment by davidkpiano** [comment]
Check the second `options` argument and make sure your actions are in `actions`:

```js
{
  guards,
  actions: {
    startCounting: assign({ startCount: context => context.count }),
    reset: assign({ count: context => context.startCount }),
    countdown: assign({ count: context => context.count - 1 })
  }
})

**Reply by pke** [reply]
Right, that was a copy&paste error. The actions where actually at the top of the file but the visualizer complained about `actions` already being defined.

However, when you run the chart in https://xstate.js.org/viz/?code=0c13d2e2de2cd30bbc2b&gist=9ce4bb4840c35b4c3567adf98dacf8fb now you can see, that its counting down, but also the progress in the timer event is only filled once and then stays filled. This seems like either a bug in the visualiser or the my chart. What do you think?

**Reply by davidkpiano** [reply]
That's a bug in the visualizer.

**Reply by pke** [reply]
Any idea what could be the cause of that? Like its not resetting its progress somehow.

**Reply by davidkpiano** [reply]
It's a CSS animation, so with the current logic, the animation is told to "play again", but replaying an animation isn't a CSS feature, so it just keeps playing the old one.
